CarpetX/src/schedule.cxx

```diff
@@ -1716,6 +1717,27 @@ int CallFunction(void *function, cFunctionData *restrict attribute,
 
   assert(active_levels);
 
+  if (CCTK_EQUALS(presync_mode, "presync-only")) {
+    const vector<clause_t> &reads = decode_clauses(attribute, rdwr_t::read);
+    std::set<int> sync_set;
+    for (const auto &rd : reads) {
+      if (CCTK_GroupTypeI(rd.gi) == CCTK_GF) {
+
+        active_levels->loop([&](const auto &restrict leveldata) {
+          const auto &restrict groupdata = *leveldata.groupdata.at(rd.gi);
+          const valid_t &need = rd.valid;
+          valid_t have = groupdata.valid.at(rd.tl).at(rd.vi).get();
+          if (need.valid_ghosts && !have.valid_ghosts && have.valid_int)
+            sync_set.insert(rd.gi);
+        });
+      }
+    }
+    if (sync_set.size() > 0) {
```

Comment:
Nitpicking: Better call `sync_set.empty()` instead of `size() > 0`.

Reply:
Fixed.